tests/unit/sync/test_sequence_tracker.py

```diff
@@ -0,0 +1,23 @@
+from unittest.mock import patch
+
+from neptune_scale.sync.sequence_tracker import SequenceTracker
+
+
+def test_update_sequence_id():
+    with patch("time.time", return_value=123.456):
+        tracker = SequenceTracker()
+
+        # Update with a positive sequence ID
+        tracker.update_sequence_id(5)
+        assert tracker.last_sequence_id == 5
+        assert tracker.last_timestamp == 123.456
+
+        # Update with a higher sequence ID
+        tracker.update_sequence_id(10)
+        assert tracker.last_sequence_id == 10
+        assert tracker.last_timestamp == 123.456
+
+        # Update with a lower sequence ID
```

Comment:
Can this actually happen?

Reply:
without lock in attribute store: yes. 
We can add lock here: 
```        
        operations = list(splitter)
        sequence_id = self._operations_repo.save_update_run_snapshots(operations)

        self._sequence_tracker.update_sequence_id(sequence_id)
```